### PR TITLE
Shadow DistSQL : drop rule statement  &  alter shadow algorithm statement

### PR DESCRIPTION
**Pull request by lanchengx**
Changes proposed in this pull request:
- add `drop shadow rule` statement
- add  related tests
- add `alter shadow algorithm` statement
- add  related tests

### CODE REVIEW

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-distsql/shardingsphere-shadow-distsql-parser/src/main/antlr4/imports/shadow/Literals.g4 [hunk 1]
```diff
@@ -25,7 +25,7 @@ STRING
     ;
 
 IDENTIFIER
-    : [A-Za-z_$0-9]*?[A-Za-z_$]+?[A-Za-z_$0-9]*
+    : [A-Za-z_$0-9-]*?[A-Za-z_$-]+?[A-Za-z_$0-9-]*
     | BQ ~'`'+ BQ
```
Comment (lanchengx) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-distsql/shardingsphere-shadow-distsql-parser/src/main/antlr4/imports/shadow/Literals.g4:29]: In order to identify the name with a'-', such as ‘user-id-simple-note-algorithms’.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-distsql/shardingsphere-shadow-distsql-parser/src/main/java/org/apache/shardingsphere/shadow/distsql/parser/core/ShadowDistSQLStatementVisitor.java [hunk 2]
```diff
@@ -65,16 +69,26 @@ public ASTNode visitShadowAlgorithmDefinition(final ShadowAlgorithmDefinitionCon
         return new ShadowAlgorithmSegment(algorithmName, algorithmSegment);
     }
     
+    private Properties getAlgorithmProperties(final AlgorithmPropertiesContext ctx) {
+        Properties result = new Properties();
+        ctx.algorithmProperty().forEach(each -> result.put(new IdentifierValue(each.key.getText()).getValue(), new StringLiteralValue(each.value.getText()).getValue()));
+        return result;
+    }
+    
     @Override
```
Comment (lanchengx) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-distsql/shardingsphere-shadow-distsql-parser/src/main/java/org/apache/shardingsphere/shadow/distsql/parser/core/ShadowDistSQLStatementVisitor.java:78]: The called method does not follow the calling method

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-distsql/shardingsphere-shadow-distsql-parser/src/main/java/org/apache/shardingsphere/shadow/distsql/parser/core/ShadowDistSQLStatementVisitor.java [hunk 3]
```diff
@@ -65,16 +69,26 @@ public ASTNode visitShadowAlgorithmDefinition(final ShadowAlgorithmDefinitionCon
...
     public ASTNode visitAlterShadowRule(final AlterShadowRuleContext ctx) {
         List<ShadowRuleSegment> shadowRuleSegments = ctx.shadowRuleDefinition().stream().map(this::visit).map(each -> (ShadowRuleSegment) each).collect(Collectors.toList());
         return new AlterShadowRuleStatement(shadowRuleSegments);
     }
     
-    private Properties getAlgorithmProperties(final AlgorithmPropertiesContext ctx) {
-        Properties result = new Properties();
-        ctx.algorithmProperty().forEach(each -> result.put(new IdentifierValue(each.key.getText()).getValue(), new StringLiteralValue(each.value.getText()).getValue()));
-        return result;
```
Comment (lanchengx) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-distsql/shardingsphere-shadow-distsql-parser/src/main/java/org/apache/shardingsphere/shadow/distsql/parser/core/ShadowDistSQLStatementVisitor.java:77]: The called method does not follow the calling method

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/distsql/exception/rule/RequiredAlgorithmMissedException.java [hunk 4]
```diff
@@ -33,4 +33,8 @@ public RequiredAlgorithmMissedException(final String schemaName) {
     public RequiredAlgorithmMissedException(final String schemaName, final Collection<String> algorithmNames) {
         super(1115, String.format("Sharding algorithms `%s` do not exist in schema `%s`.", algorithmNames, schemaName));
```
Comment (taojintianxia) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/distsql/exception/rule/RequiredAlgorithmMissedException.java:34]: what does 1115 mean . is there any constant or enum to describe this ?

Comment (lanchengx) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/distsql/exception/rule/RequiredAlgorithmMissedException.java:34]: This is `errorCode`, the same exception class `errorCode` is the same, 1115 is used directly in the current class, so I keep the same, but I did not find an explanation for 1115.

**Comment by lanchengx** [comment]
@tristaZero  hi, could you have a review?

**Review by tristaZero — APPROVED** [review]

**Pull request merged**